libherokubuildpack/src/write.rs

```diff
@@ -57,10 +62,44 @@ pub struct TeeWrite<A: io::Write, B: io::Write> {
     inner_b: B,
 }
 
-impl<W: io::Write> MappedWrite<W> {
+impl<W> MappedWrite<W>
+where
+    W: io::Write,
+{
+    pub fn new<F: (Fn(Vec<u8>) -> Vec<u8>) + Sync + Send + 'static>(
```

Comment:
Should `new` instead be private? Previously it wasn't possible to create `MappedWrite` manually (only via `mapped()`) since there was no `MappedWrite::new` and the `MappedWrite` struct fields are private.

```suggestion
    pub new<F: (Fn(Vec<u8>) -> Vec<u8>) + Sync + Send + 'static>(
```

Reply:
Or to put it another way: If `MappedWrite::new` is now the recommended way for end users to construct a `MappedWrite` struct, then why do we still also have `mapped()`? For backwards compat and we'll remove later? (In which case perhaps we just remove now?) 

Or else, if `MappedWrite::new` isn't the recommended way for end users to construct a `MappedWrite` (and we still want them to use `mapped()`, then it seems `MappedWrite::new` should not be `pub`.

Reply:
Yeah, it should not be `pub` here. I created it so that the weirdness with the `Option` wrapping is contained in one area as much as possible. Since we have `mapped` and `line_mapped` that configure `MappedWriter` in certain ways, using those is preferred. 

I removed the `pub` in [934ab1f](https://github.com/heroku/libcnb.rs/pull/765/commits/934ab1f3263f4f313880d56f3fd9d253a76c9d99).